library/core/src/intrinsics/mod.rs

```diff
@@ -362,7 +362,38 @@ pub fn rustc_peek<T>(_: T) -> T;
 /// on most platforms.
 /// On Unix, the
 /// process will probably terminate with a signal like `SIGABRT`, `SIGILL`, `SIGTRAP`, `SIGSEGV` or
-/// `SIGBUS`.  The precise behavior is not guaranteed and not stable.
+/// `SIGBUS`.
+/// On Windows, the
+/// process will use `__fastfail` to terminate the process, which will terminate the process
+/// immediately without running any in-process exception handlers. In earlier versions of Windows,
+/// this sequence of instructions will be treated as an access violation, which
+/// will still terminate the process but might run some exception handlers.
+///
+/// The precise behavior is not guaranteed and not stable.
```

Comment:
This entire doc comment only applies to the Windows version of this function. So doc.rust-lang.org would no longer show it.